python/graphstorm/dataloading/dataset.py

```diff
@@ -149,24 +160,52 @@ def __init__(self, graph_name, part_config, node_feat_field, edge_feat_field):
-                        f" no such node type."
-                data = {}
-                feat_names = node_feat_field[ntype]
-                for name in feat_names:
-                    data[name] = self.load_wg_feat(part_config, num_parts, ntype, name)
-                if len(self._g.ntypes) == 1:
-                    self._g._ndata_store.update(data)
-                else:
-                    self._g._ndata_store[ntype].update(data)
-
+            if node_feat_field:
+                if isinstance(node_feat_field, str):
+                    node_feat_field = {ntype: [node_feat_field] for ntype in self._g.ntypes}
+                for ntype, feat_names in node_feat_field.items():
+                    data = {}
+                    for name in feat_names:
+                        wg_folder = os.path.join(os.path.dirname(part_config), 'wholegraph')
+                        assert len([feat_file for feat_file in os.listdir(wg_folder) \
+                            if re.search(ntype + '~' + name, feat_file)]) > 0, \
+                            f"Feature '{name}' of '{ntype}' is not in WholeGraph format. " \
+                            f"Please convert all the available features to WholeGraph " \
+                            f"format to utilize WholeGraph."
+                        data[name] = self.load_wg_feat(part_config, num_parts, ntype, name)
+                    if len(self._g.ntypes) == 1:
+                        self._g._ndata_store.update(data)
+                    else:
+                        self._g._ndata_store[ntype].update(data)
+
+            # load edge feature from wholegraph memory
+            # TODO(IN): Add support for edge_feat_field
+            if decoder_edge_feat:
+                if isinstance(decoder_edge_feat, str):
+                    decoder_edge_feat = {etype: [decoder_edge_feat] \
+                        for etype in self._g.canonical_etypes}
+                for etype, feat_names in decoder_edge_feat.items():
+                    data = {}
+                    etype_wg = ":".join(etype)
+                    for name in feat_names:
+                        wg_folder = os.path.join(os.path.dirname(part_config), 'wholegraph')
+                        assert len([feat_file for feat_file in os.listdir(wg_folder) \
+                            if re.search(etype_wg + '~' + name, feat_file)]) > 0, \
+                            f"Feature '{name}' of '{etype}' is not in WholeGraph format. " \
+                            f"Please convert all the available features to WholeGraph " \
+                            f"format to utilize WholeGraph."
+                        data[name] = self.load_wg_feat(part_config, num_parts, etype_wg, name)
+                    if len(self._g.canonical_etypes) == 1:
+                        self._g._edata_store.update(data)
+                    else:
+                        self._g._edata_store[etype].update(data)
+
+        barrier()
```

Comment:
should this barrier() within the `if is_distributed() and is_wholegraph():` code block?